src/main/scala/com/fortysevendeg/android/sqlite/resultset/SQLDroidResultSet.scala

```diff
@@ -0,0 +1,688 @@
+package com.fortysevendeg.android.sqlite.resultset
+
+import java.io._
+import java.math.BigDecimal
+import java.net.URL
+import java.sql._
+import java.text.SimpleDateFormat
+import java.util
+import java.util.Calendar
+
+import android.database.Cursor
+import com.fortysevendeg.android.sqlite._
+
+import scala.util.{Failure, Success, Try}
+import scala.{Array => SArray}
+
+class SQLDroidResultSet(
+  cursor: Cursor,
+  log: LogWrapper = new AndroidLogWrapper())
+  extends ResultSet
+  with NonEditableResultSet
+  with WrapperNotSupported {
+
+  private[this] var lastColumnRead: Option[Int] = None
+
+  private[this] def moveOp(f: => Boolean): Boolean =
+    f match {
+      case true =>
+        lastColumnRead = None
+        true
+      case false =>
+        false
+    }
+
+  private[this] def getOp[T](index: Int)(f: Int => T): T = {
+    lastColumnRead = Some(index)
+    f(index)
+  }
+
+  private[this] val dateFormat = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss.S")
+
+  private[this] def parseDate(date: String): Option[java.util.Date] =
+    Try(dateFormat.parse(date)) match {
+      case Success(d) => Some(d)
+      case Failure(e) =>
+        log.e("Can't parse date", Some(e))
+        None
+    }
+
+  private[this] def notClosed[T](f: => T) =
+    if (cursor.isClosed) throw new SQLException("Cursor is closed")
+    else Try(f) match {
+      case Success(r) => r
+      case Failure(e) => throw new SQLException(e)
+    }
+
+  private[this] def notNull[T](columnIndex: Int)(f: => T): Option[T] =
+    getOp(columnIndex.index)(cursor.isNull) match {
+      case true => None
+      case false => Option(f)
+    }
+
+  private[this] def readBytes(columnIndex: Int): Option[SArray[Byte]] =
+    notNull(columnIndex) {
+      cursor.getTypeSafe(columnIndex.index) match {
+        case Cursor.FIELD_TYPE_STRING => getOp(columnIndex.index)(cursor.getString).getBytes
```

Comment:
Just be sure you have an stable identifier: http://stackoverflow.com/questions/7083502/why-cant-a-variable-be-a-stable-identifier